### DISCUSSION TITLE
Word wrap should favor not breaking words unless desperate

### DISCUSSION THREAD

**Discussion by jsoref** [community/community#45069] [Mobile] [answered]
### Select Topic Area

Bug

### Body

GitHub Mobile beta

1. Open https://github.com/actions/cache/pull/1035
2. Tap `32 files changed` – note that the order of the next 4 steps matters a lot – it shouldn't, but I'll file that next
3. Tap the ⚙️ 
4. Enable `wrap lines`
5. Tap 🔙
6. Collapse `.devcontainer/devcontainer.json`... `package.json`
7. Scroll down a bit and tap line 7 (and 8) because it looks like `wild `+`card`:
    ![Screenshot_20230122-105046](https://user-images.githubusercontent.com/2119212/213925402-1ad948cb-eec2-4c0c-96cb-fe8f85c61cfa.png)

### Expected results
Unless desperate, don't split a word across lines.

Desperate means that if there is only one word on the line.

Also, when desperately splitting a solitary word, including a magical soft-wrap character ↩️ would be appreciated.

### Actual results
Both `wildcard ` and `for` were split.

**Comment by iskae** [comment, marked as answer]
Hi @jsoref 👋 

When the word wrapping is enabled through code options we are simply letting the OS handle the word wrapping when the width of the cell exceeds the width of the screen. We can at most set a wrapping strategy for it, but unfortunately we are lacking a more fine-grained control over it due to limitations 😅 

I will be forwarding your request for adding a soft-wrap character at the end of the line to our designers though 👍 

Thanks once again for your suggestion 🙇

**Reply by jsoref** [reply]
https://github.com/cert-manager/cert-manager/pull/7748/files

Here, `their` is needlessly split into `the`+`ir`:
![`their` split across lines as `the`+`ir`](https://github.com/user-attachments/assets/76134a21-c265-4d3c-9e66-44170571a2d0)
